# Data Preparation

## Dataset Overview

### Data Source
Los Angeles restaurant data from public sources.

### Dataset Description

**Table: rest_data**

| Column | Description |
|--------|-------------|
| `id` |  Restaurant establishment unique id |
| `object_name` |  Restaurant establishment name |
| `address` | Full address of the establishment |
| `chain` | Whether the establishment is part of a chain (TRUE/FALSE) |
| `object_type` | Type of establishment (e.g., restaurant, cafe, bar) |
| `number` | Number of seats in the establishment |


## Objectives
The main objective of this notebook/steps is to verify data integrity and correct possible anomalies. Data will be loaded and some verifications will be performed to ensure the best quality for EDA.

1. **Load** the restaurant data from the CSV file
2. **Validate** data types for each column
3. **Check** for missing values and duplicates
4. **Process** the data if necessary
5. **Ensure** data quality for subsequent exploratory analysis (EDA)

In [1]:
# Importing necessary libraries
import pandas as pd

In [2]:
# Loading the dataset
df_restaurants = pd.read_csv("../data/rest_data_us_upd.csv")

In [3]:
# Checking dataframe info
df_restaurants.info()

<class 'pandas.DataFrame'>
RangeIndex: 9651 entries, 0 to 9650
Data columns (total 6 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   id           9651 non-null   int64 
 1   object_name  9651 non-null   str   
 2   address      9651 non-null   str   
 3   chain        9648 non-null   object
 4   object_type  9651 non-null   str   
 5   number       9651 non-null   int64 
dtypes: int64(2), object(1), str(3)
memory usage: 452.5+ KB


In [4]:
# Checking df first 5 rows
df_restaurants.head(5)

,id,object_name,address,chain,object_type,number
0,11786,HABITAT COFFEE SHOP,3708 N EAGLE ROCK BLVD,False,Cafe,26
1,11787,REILLY'S,100 WORLD WAY 120,False,Restaurant,9
2,11788,STREET CHURROS,6801 HOLLYWOOD BLVD 253,False,Fast Food,20
3,11789,TRINITI ECHO PARK,1814 W SUNSET BLVD,False,Restaurant,22
4,11790,POLLEN,2100 ECHO PARK AVE,False,Restaurant,20


- Dataframe have a total of 9651 rows;
- No null values in columns except for'chain' column that have 3 NaN; due to these NaN values variable type is not correct (should be 'boolean');
- Column 'object_type' can possible be a categorical variable; content need to be cheked;
- Other columns are already in correct type.

In [5]:
# Drop rows with NaN values in 'chain' column
df_restaurants.dropna(subset=['chain'], inplace=True)

In [6]:
# Converting column 'chain' to boolean type
df_restaurants['chain'] = df_restaurants['chain'].astype(bool)

In [7]:
# Checking 'chain' distribution
df_restaurants['chain'].value_counts(normalize=True)

chain
False    0.618988
True     0.381012
Name: proportion, dtype: float64

In [8]:
# Checking duplicated rows in df
dup_rows = df_restaurants.duplicated().sum()
print(f'df_restaurants have {dup_rows} duplicated rows')

df_restaurants have 0 duplicated rows


In [9]:
# Checking id column integrity
unique_id = df_restaurants['id'].nunique()
print(f'id column have {unique_id} unique values')

id column have 9648 unique values


In [10]:
# Checking names unique values
df_restaurants['object_name'].nunique()

8669

In [11]:
# Checking establishments options
df_restaurants['object_type'].value_counts()

object_type
Restaurant    7253
Fast Food     1066
Cafe           435
Pizza          319
Bar            292
Bakery         283
Name: count, dtype: int64

In [12]:
# Converting 'object_type' to category
df_restaurants['object_type'] = df_restaurants['object_type'].astype(
    'category')

In [13]:
# Checking establishments number of seats distribution
df_restaurants['number'].describe()

count    9648.000000
mean       43.703566
std        47.627252
min         1.000000
25%        14.000000
50%        27.000000
75%        46.000000
max       229.000000
Name: number, dtype: float64

In [14]:
# Checking duplicated values in 'address' column
dup_address = df_restaurants['address'].duplicated().sum()
print(f'df_restaurants have {dup_address} duplicated addresses')

df_restaurants have 1139 duplicated addresses


In [15]:
# Checking duplicated values in 'address' column
df_restaurants[df_restaurants['address'].duplicated(
    keep=False)].sort_values('address').head(10)

,id,object_name,address,chain,object_type,number
5624,17410,SOUPLANTATION # 17,100 N LA CIENEGA BLVD E-3,True,Restaurant,199
1588,13374,SOUPLANTATION,100 N LA CIENEGA BLVD E-3,True,Restaurant,181
6277,18063,GORDEN BIERSCH,100 WORLD WAY,False,Restaurant,17
8201,19987,HOME TURF SPORTS BAR,100 WORLD WAY,False,Bar,32
6515,18301,STARBUCKS COFFEE,100 WORLD WAY,True,Cafe,9
5989,17775,LA BREA BAKERY,100 WORLD WAY,True,Bakery,26
5996,17782,TORN BASIL,100 WORLD WAY,False,Restaurant,5
6000,17786,PINKBERRY #78,100 WORLD WAY,True,Restaurant,3
7509,19295,O'HARA'S,1000 GAYLEY AVE,False,Restaurant,63
8946,20732,ROCCO'S TAVERN,1000 GAYLEY AVE,True,Restaurant,134


In [16]:
# Final dataframe validation
df_restaurants.info()

<class 'pandas.DataFrame'>
Index: 9648 entries, 0 to 9650
Data columns (total 6 columns):
 #   Column       Non-Null Count  Dtype   
---  ------       --------------  -----   
 0   id           9648 non-null   int64   
 1   object_name  9648 non-null   str     
 2   address      9648 non-null   str     
 3   chain        9648 non-null   bool    
 4   object_type  9648 non-null   category
 5   number       9648 non-null   int64   
dtypes: bool(1), category(1), int64(2), str(2)
memory usage: 395.8 KB


### Observations
- 'id' column has no duplicated values, which confirms data integrity as this should be a unique identifier;
- 'object_name' has a total of 8669 unique names out of 9648 rows, meaning 979 names appear more than once — these could be chain establishments or independent restaurants that share the same name; further analysis needed in EDA;
- 'object_type' distribution shows most of the data refers to 'Restaurants' (7253), followed by 'Fast Food' (1066) and then 'Cafe' (435), 'Pizza' (319) and 'Bakery' (283);
- 'number' column has a correct range, between 1 and 229 seats; looking at the mean of 43.70 and median of 27 we have a right-skewed distribution, possibly due to some higher values (e.g., max is 229 seats);
- 'address' column has 1139 duplicated values; we see a street with 2 restaurants and '100 WORLD WAY' looks like a shopping center as we have several types of establishments.

In [17]:
# Saving dataframe to a new file
df_restaurants.to_csv("../data/rest_data_cleaned.csv", index=False)